internal/helm/chart/builder_remote.go

```diff
@@ -186,7 +186,6 @@ func (b *remoteChartBuilder) downloadFromRepository(remote *repository.ChartRepo
 		err = fmt.Errorf("could not load repository index for remote chart reference: %w", err)
 		return nil, &BuildError{Reason: ErrChartPull, Err: err}
 	}
-	defer remote.Unload()
```

Comment:
With the removal of Unload(), my recommendation in https://github.com/fluxcd/source-controller/pull/690/files#r878152484 becomes more feasible.

Reply:
Yes 👍 I am taking your recommendation into account in the coming PR that enable dependencies from OCI Helm Repositories.